backend/app/sql/total_pop_udf.sql

```diff
@@ -0,0 +1,24 @@
+CREATE OR REPLACE FUNCTION get_total_population(document_id UUID)
+RETURNS TABLE (zone TEXT, total_pop BIGINT) AS $$
+DECLARE
+    gerrydb_table_name TEXT;
+    sql_query TEXT;
+BEGIN
+    SELECT gerrydb_table INTO gerrydb_table_name
+    FROM document
+    WHERE document.document_id = $1;
+
+    sql_query := format('
+        SELECT
+            assignments.zone::TEXT AS zone,
+            SUM(blocks.total_pop)::BIGINT AS total_pop
```

Comment:
(o) we should think about how we want to handle unmatched geoids- if we're going to cast nulls to zeros upstream and we aren't doing anything on the backend with the nulls, we should just do that in the sql query. If we still need nulls for geoids with missing population information, then this is fine as written.

Reply:
Ohhh good point – for now the right answer is probably to cast nulls as zeros in sql query.

Raises another question: do we need a constraint on inserted `geoids` that they're in the document's `gerrydb_table`? This would probably be slow / cumbersome but we probably want to catch unmatched geoids before we get here. unmatched `geoids` could come from: (1) user manually edits blob sent over wire, (2) gerrydb view lost/corrupted, (3) gerrydb updated by us

Reply:
hm, where do candidate geoids come from in a document? IMO ideally this is a one-time validation we do when something changes in either the geoids we make available to paint or on gerrydb load.

Reply:
For now I `SUM(COALESCE(total_pop), 0))`